VS2017 64bit doesn't compile v0.3.0 with UA_ENABLE_DISCOVERY_MULTICAST

<!--
!ATTENTION!
Please read the following page carefully and provide us with all the
information requested:
https://github.com/open62541/open62541/wiki/Writing-Good-Issue-Reports

Use Github Markdown to format your text:
https://help.github.com/articles/basic-writing-and-formatting-syntax/

Fill out the sections and checklist below (add text at the end of each line).

!ATTENTION!
--------------------------------------------------------------------------------
-->

## Description

I'm trying to compile the open62541-library using Visual Studio 2017 64bit with the DiscoveryMulticast-Feature enabled. When I run *Build* on the *ALL_BUILD*-project I run into the following error:

`<path-to>\open62541\src\server\ua_services_discovery_multicast.c(341):
 error C2220: warning treated as error - no 'object' file generated `
`<path-to>\open62541\src\server\ua_services_discovery_multicast.c(341): warning C4244: 'function': conversion from 'SOCKET' to 'int', possible loss of data`
`<path-to>\open62541\src\server\ua_services_discovery_multicast.c(631): warning C4244: 'function': conversion from 'SOCKET' to 'int', possible loss of data`

Because of this error the building of the *open62541-object*-project failes and the linker later sais

`9>LINK : fatal error LNK1181: cannot open input file '<path-to>\open62541\build\open62541-object.dir\Debug\ua_services_discovery_multicast.obj'
`

in the end the build process is summarized with

`========== Build All: 8 succeeded, 2 failed, 0 skipped ==========`

## Background Information / Reproduction Steps

I used CMake 3.12.0-rc2 (cmake-gui) for Windows:

1. checked out the branch 0.3 with the submodules
1. Selected \<path-to>/open62541 as "Where is the source code:"
1. Selected \<path-to>/open62541/build as "Where to build the binaries:"
1. clicked *Configure*
1. selected "Visual Studio 15 2017 Win64" to specify the generator for this project
1. enabled the checkbox at "UA_ENABLE_DISCOVERY_MULTICAST"
1. clicked *Generate*
1. opened the Project
1. right-clicked on the *ALL_BUILD*-project in the *Solution Explorer*
1. clicked *Build* in the context menu

## Checklist

 - [x] open62541 Version (release number or git tag): I checked out the [0.3 branch](https://github.com/open62541/open62541/tree/0.3)
 - [x] Other OPC UA SDKs used (client or server): nope
 - [x] Operating system: Windows 10 Pro 64 bit (Version 1803)
 - Logs (with `UA_LOGLEVEL` set as low as necessary) attached: not generated
 - Wireshark network dump attached: not necessary
 - Self-contained code example attached: not necessary
 - [ ] Critical issue

--- a/arch/network_tcp.c
+++ b/arch/network_tcp.c
@@ -119,8 +119,14 @@ connection_recv(UA_Connection *connection, UA_ByteString *response,
         return UA_STATUSCODE_BADOUTOFMEMORY; /* not enough memory retry */
     }
 
-    size_t offset = connection->incompleteChunk.length;
-    size_t remaining = connection->config.recvBufferSize - offset;
+#ifdef _WIN32
+    // windows requires int parameter for length
+    int offset = (int)connection->incompleteChunk.length;
+    int remaining = connection->config.recvBufferSize - offset;
+#else
+	size_t offset = connection->incompleteChunk.length;
+	size_t remaining = connection->config.recvBufferSize - offset;
+#endif
 
     /* Get the received packet(s) */
     ssize_t ret = UA_recv(connection->sockfd, (char*)&response->data[offset],

--- a/src/server/ua_services_discovery_multicast.c
+++ b/src/server/ua_services_discovery_multicast.c
@@ -494,7 +494,7 @@ iterateMulticastDiscoveryServer(UA_Server* server, UA_DateTime *nextRepeat,
                                 UA_Boolean processIn) {
     struct timeval next_sleep = { 0, 0 };
     unsigned short retval = mdnsd_step(server->discoveryManager.mdnsDaemon,
-                                       server->discoveryManager.mdnsSocket,
+                                       (int)server->discoveryManager.mdnsSocket,
                                        processIn, true, &next_sleep);
     if(retval == 1) {
         UA_LOG_SOCKET_ERRNO_WRAP(

